core/domain/collection_domain.py

```diff
@@ -425,10 +505,12 @@ def to_yaml(self):
         # YAML representation.
         del collection_dict['id']
 
-        return utils.yaml_from_dict(collection_dict)
+        return utils.yaml_from_dict(collection_dict) # type: ignore[arg-type]
```

Comment:
Why is this ignore needed?

Reply:
`utils.yaml_from_dict` is a general function which accepts dict[str, any], but here collection_dict is typed with typeddict . so, thats why i added it.

Reply:
Okay, add a code comment explaining it.

Reply:
Done!